kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala

```diff
@@ -49,7 +54,10 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
   }
 
   private def startInternal(): Unit = {
-    server.addHandler(ApiRootResource.getServletHandler(this))
+    val contextHandler = ApiRootResource.getServletHandler(this)
+    val holder = new FilterHolder(new AuthenticationFilter(conf))
+    contextHandler.addFilter(holder, "/*", EnumSet.allOf(classOf[DispatcherType]))
```

Comment:
Do we need to add the authentacation filter to "/*"? Maybe we need to give some exceptions, e.g. "/healthy-check", "/metrics/**"